tests/integration/utils.py

```diff
@@ -463,3 +465,109 @@ def _validate_expected_dep_file_contents(dep_contents_file: Path, output_dir: Pa
         dep_file = output_dir / "deps" / path
         assert dep_file.exists()
         assert dep_file.read_text() == expected_content
+
+
+TEST_EVERYTHING = object()
+TEST_NOTHING = object()
+# Individual files could be added to the set as well.
+paths_to_code = set((Path("cachi2"), Path("tests/integration")))
+supported_pms = set(list(resolver._package_managers) + list(resolver._dev_package_managers))
+
+
+def retrieve_changed_files_from_git(ref: str = "HEAD") -> tuple[Path, ...]:
+    repo = Repo(".", search_parent_directories=True)
+    commit = repo.commit(ref)
+    modified_files = [Path(item.a_path) for item in commit.diff(commit.parents)]  # type: ignore
```

Comment:
What did mypy complain about that you have to disable it?
For future reference I think it's worth adding a commentary to all places where mypy needs to be disabled and why.

Reply:
> When invoked this works fine now because deep inside the library [args sequence gets flattened](https://github.com/gitpython-developers/GitPython/blob/cfadd9e5edb8571c2c15878c8eeca91f15a69040/git/cmd.py#L1477) and also because all normal commits will have exactly one parent. OTOH multiparent commits are merge commits, and IIRC they will not have any changes in them anyway (unless they are evil), so they should return nothing and it appears to be safe to accept multiple parents. It looks like the code does the right thing while annotations miss this aspect.

Yeah, another way of fixing this would be `commit.parents[0]` because in case of a merge commit parent, that one should always point to the other branch which we merged to the target branch and so would become nr.2 parent. I completely agree though that stats seems like the superior approach.